### PR TITLE
#1054 - fix crash and UI

### PR DESCRIPTION
**Pull request by nullkiller**

### CODE REVIEW

File: client/windows/CreaturePurchaseCard.cpp [hunk 1]
```diff
@@ -98,21 +98,20 @@ void CreaturePurchaseCard::initView()
...
 }
 
 CreaturePurchaseCard::CCreatureClickArea::CCreatureClickArea(const Point & position, const std::shared_ptr<CCreaturePic> creaturePic, const CCreature * creatureOnTheCard)
 	: CIntObject(RCLICK),
 	creatureOnTheCard(creatureOnTheCard)
 {
-	pos.x = position.x;
-	pos.y = position.y;
+	pos.x += position.x;
+	pos.y += position.y;
```
Comment (nullkiller) [inline client/windows/CreaturePurchaseCard.cpp:114]: Gives relative coords

File: client/windows/CreaturePurchaseCard.cpp [hunk 2]
```diff
@@ -47,8 +47,8 @@ void CreaturePurchaseCard::switchCreatureLevel()
 	auto index = vstd::find_pos(upgradesID, creatureOnTheCard->idNumber);
 	auto nextCreatureId = vstd::circularAt(upgradesID, ++index);
 	creatureOnTheCard = nextCreatureId.toCreature();
-	creatureClickArea = std::make_shared<CCreatureClickArea>(Point(pos.x + CCreatureClickArea::CREATURE_X_POS, pos.y + CCreatureClickArea::CREATURE_Y_POS), picture, creatureOnTheCard);
 	picture = std::make_shared<CCreaturePic>(parent->pos.x, parent->pos.y, creatureOnTheCard);
+	creatureClickArea = std::make_shared<CCreatureClickArea>(Point(pos.x, pos.y), picture, creatureOnTheCard);
```
Comment (nullkiller) [inline client/windows/CreaturePurchaseCard.cpp:51]: @AgostonSzepessy This still does not work

Comment (nullkiller) [reply client/windows/CreaturePurchaseCard.cpp]: Fixed as well.

**Review by kambala-decapitator — APPROVED** [review]
thanks, works great now!

**Pull request merged**